src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateAtTheMoment = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    let stateCopy = { ...stateAtTheMoment };
+
+    switch (action.type) {
+      case 'clear':
+        stateCopy = {};
```

Comment:
Better way here will be or to use delete in loop for all properties or to use Object.assign to an empty object